plugins/modules/identity/keycloak/keycloak_clientsecret.py

```diff
@@ -0,0 +1,236 @@
+    keycloak_argument_spec, get_token, KeycloakError
+from ansible.module_utils.basic import AnsibleModule
+
+
+def main():
+    """
+    Module execution
+
+    :return:
+    """
+    argument_spec = keycloak_argument_spec()
+
+    meta_args = dict(
+        state=dict(default='present', choices=['present', 'absent']),
+        realm=dict(default='master'),
+
+        id=dict(type='str'),
+        client_id=dict(type='str', aliases=['clientId']),
+    )
+
+    argument_spec.update(meta_args)
+
+    module = AnsibleModule(argument_spec=argument_spec,
+                           supports_check_mode=True,
+                           required_one_of=([['id', 'client_id'],
+                                             ['token', 'auth_realm', 'auth_username', 'auth_password']]),
+                           required_together=([['auth_realm', 'auth_username', 'auth_password']]))
+
+    result = dict(changed=False, msg='', diff={}, clientsecret='')
+
+    # Obtain access token, initialize API
+    try:
+        connection_header = get_token(module.params)
+    except KeycloakError as e:
+        module.fail_json(msg=str(e))
+
+    kc = KeycloakAPI(module, connection_header)
+
+    realm = module.params.get('realm')
+    state = module.params.get('state')
+    id = module.params.get('id')
+    client_id = module.params.get('client_id')
+
+    # only lookup the client_id if id isn't provided.
+    # in the case that both are provided, prefer the ID, since it's one
+    # less lookup.
+    if id is None and client_id is not None:
+        client = kc.get_client_by_clientid(client_id, realm=realm)
+
+        if client is None:
+            raise Exception('Client does not exist {client_id}'.format(client_id=client_id))
+
+        id = client['id']
+
+    if state == 'present':
+        # Get secret
+        result['changed'] = False
+
+        if module.check_mode:
+            module.exit_json(**result)
+
+        # Create new secret
+        clientsecret = kc.get_clientsecret(id=id, realm=realm)
+
+        result['clientsecret'] = clientsecret
+        result['end_state'] = clientsecret
+        result['msg'] = 'Get client secret successful for ID {id}'.format(id=id)
+
+        module.exit_json(**result)
+    else:
+        if state == 'absent':
+            # Process a creation
+            result['changed'] = True
+
+            if module.check_mode:
+                module.exit_json(**result)
+
+            # Create new secret
+            clientsecret = kc.create_clientsecret(id=id, realm=realm)
+
+            result['end_state'] = clientsecret
+            result['msg'] = 'New client secret has been generated for ID {id}'.format(id=id)
+
+            module.exit_json(**result)
+
+        # Do nothing and exit
+        result['changed'] = False
+        result['end_state'] = {}
+        result['msg'] = 'State not specified; doing nothing.'
+        module.exit_json(**result)
```

Comment:
This line is redundant, as the exact same command will run after the `if-else` block in line 232